runtime/near-vm-logic/src/logic.rs

```diff
@@ -2670,6 +2702,19 @@ pub struct VMOutcome {
     pub logs: Vec<String>,
     /// Data collected from making a contract call
     pub profile: ProfileData,
+    action_receipts: ActionReceipts,
```

Comment:
Dual observation to that about `epoch_manager`: in this struct, all fields are `pub` -- it's a plain old data, value type with defined identity. Adding a private field here breaks valueness. 

I think it'll actually be ok to make `ActionReceipts` public, and introduce some kind of `VMLogicRecipt`. 
take_receipts
That is, rather than doing `VMOutcome::take_receipts`, the caller would themselves convert VMLogicRecipts to proper Receipts.

Reply:
Sounds good! I was trying hard to make it so there wasn't breaking changes and exposing things that aren't needed by nearcore. Apparently too careful :)

I'll make this change now and open this up for review

Reply:
>I was trying hard to make it so there wasn't breaking changes

We don't care about backwards compatability at this layer. On the contrary, we are willing to change all the call-sites to prevent tech debt from accumulating.

Reply:
The only question about this is should I update the `PartialEq` implementation for outcome to include the receipts? I avoided it for now to avoid breaking change, but don't have an opinion